src/client/Microsoft.Identity.Client/Internal/ClientCredentialWrapper.cs

```diff
@@ -19,104 +23,22 @@ namespace Microsoft.Identity.Client.Internal
-                case ConfidentialClientAuthenticationType.SignedClientAssertionDelegate:
-                    SignedAssertionDelegate = config.SignedClientAssertionDelegate;
-                    break;
-            }
-        }
-
-        #region TestBuilders
-        //The following builders methods are intended for testing
-        public static ClientCredentialWrapper CreateWithCertificate(X509Certificate2 certificate, IDictionary<string, string> claimsToSign = null)
-        {
-            return new ClientCredentialWrapper(certificate, claimsToSign);
-        }
-
-        public static ClientCredentialWrapper CreateWithSecret(string secret)
-        {
-            var app = new ClientCredentialWrapper(secret, ConfidentialClientAuthenticationType.ClientSecret);
-            app.AuthenticationType = ConfidentialClientAuthenticationType.ClientSecret;
-            return app;
-        }
-
-        public static ClientCredentialWrapper CreateWithSignedClientAssertion(string signedClientAssertion)
-        {
-            var app = new ClientCredentialWrapper(signedClientAssertion, ConfidentialClientAuthenticationType.SignedClientAssertion);
-            app.AuthenticationType = ConfidentialClientAuthenticationType.SignedClientAssertion;
-            return app;
-        }
-
-        private ClientCredentialWrapper(X509Certificate2 certificate, IDictionary<string, string> claimsToSign = null)
-        {
-            ConfidentialClientApplication.GuardMobileFrameworks();
-
-            Certificate = certificate;
-
-            if (claimsToSign != null && claimsToSign.Any())
-            {
-                ClaimsToSign = claimsToSign;
-                AuthenticationType = ConfidentialClientAuthenticationType.ClientCertificateWithClaims;
-                return;
-            }
-
-            AuthenticationType = ConfidentialClientAuthenticationType.ClientCertificate;
-        }
-
-        private ClientCredentialWrapper(string secretOrAssertion, ConfidentialClientAuthenticationType authType)
-        {
-            ConfidentialClientApplication.GuardMobileFrameworks();
-
-            if (authType == ConfidentialClientAuthenticationType.SignedClientAssertion)
-            {
-                SignedAssertion = secretOrAssertion;
-            }
-            else
-            {
-                Secret = secretOrAssertion;
-            }
-        }
-
-        #endregion TestBuilders
-        private void CheckCertificateKeySize(X509Certificate2 cert)
-        {
-            //Currently, the min key size is enforced on desktop (net 45) as 2048 as it is the current industry standard.
-            //This is not enforced on netCore unfortunately and adding this enforcement may break customers on netCore.
-            //NetCore can only enforce a min key size of 512 since it is enforced by the portal already.
-#if DESKTOP
-            if (cert.PublicKey.Key.KeySize < MinKeySizeInBits)
+            if (config.ConfidentialClientCredentialCount == 0)
```

Comment:
is there a reason to remove this from the ValidateCredentialParameters() method?

Reply:
it wasn't used anywhere and no test broke. I think this is enforced lower down the stack.